src/parser/content-evaluator-module.ts

```diff
@@ -178,15 +181,33 @@ export class ContentEvaluatorModule extends BaseModule {
       const dummyResponse = JSON.stringify(this._generateDummyResponse(comments), null, 2);
       const maxTokens = this._calculateMaxTokens(dummyResponse);
 
-      const promptForComments = this._generatePromptForComments(specification, comments, allComments);
+      let promptForComments = this._generatePromptForComments(specification, comments, allComments);
+      if (this._calculateMaxTokens(promptForComments, Infinity) > TOKEN_MODEL_LIMIT) {
```

Comment:
I'm skeptical about this whole TFIDF approach

1. The tokenizer algorithm is different per model and I'm not sure what model we should be using. I'm starting to see a pattern of OpenAI and Anthropic release cycles alternating so to get the best models perhaps we should also alternate. I heard good things about the new OpenAI model. 
2. I don't fully understand why we need this if we can just filter out a lot more overhead like the bot comments, slash commands 
3. The weaknesses of this approach isn't fully clear to me. We did some earlier research on implementing this method in here before but I recall that it wasn't a good fit. I just don't remember why- perhaps accuracy related.

Reply:
It should be an edge case for very long conversations which should not happen very often. Advantage about TFIDF is that we keep a bunch of relevant comments instead of arbitrarily keep the first / last n comments.

Reply:
Can you articulate the weaknesses or concerns

Reply:
The main concern is that we base the selection solely on similarity, which might be less precise that using the LLM to determine which comments are the most relevant. On the other hand I think it is better than actually writing summaries of comments because this way we keep the content intact during evaluation (just shorten it).

Since this should not happen often (and even less often now that we will properly filter useless comments once I resolve your [comment](https://github.com/ubiquity-os-marketplace/text-conversation-rewards/pull/225#discussion_r1899029223)), the easiest and most straightforward solution seemed to be the best choice.